### ISSUE TITLE
push to luarocks

### ISSUE THREAD

**Issue by atticus-sullivan** [lgi-devs/lgi#302]
Do you still push to luarocks? (would be really comfortable for installing the module)
https://luarocks.org/modules/pavouk/lgi is quite old and is not working for lua5.4 (which is already fixed after #247)

**Comment by Frityet** [comment]
just use 

`sudo luarocks install --server=https://luarocks.org/dev lgi`

**Comment by atticus-sullivan** [comment]
Yes already tried this, the Problem is that this way `V 0.9.2-1` will be installed which is from 2017 and has no lua5.4 support.
(What works is using the provided rocspec removing the `tag` from the `source` -> just use the current `master`, but that's not really convenient)
But I see maybe the bigger Problem is that there is no more recent version than `0.9.2-1` despite the fact it is 5 years old and there actually was some (important, if I think of lua5.4 support) development in that period of time.

**Comment by Frityet** [comment]
> Yes already tried this, the Problem is that this way `V 0.9.2-1` will be installed which is from 2017 and has no lua5.4 support. (What works is using the provided rocspec removing the `tag` from the `source` -> just use the current `master`, but that's not really convenient) But I see maybe the bigger Problem is that there is no more recent version than `0.9.2-1` despite the fact it is 5 years old and there actually was some (important, if I think of lua5.4 support) development in that period of time.

That is a big issue. Damn, maybe the dep should be updated for Lua5.4?

**Comment by atticus-sullivan** [comment]
> That is a big issue

Yes, I agree.

I see two things
1. Change (and push the change to luarocks) the deps so that one can only install this for lua versions `< lua5.4`.
2. Make a new release + rockspec (and push this to luarocks)

**Comment by leafo** [comment]
> just use
> 
> `sudo luarocks install --server=https://luarocks.org/dev lgi`

There is no development/scm rockspec uploaded to the lgi module on luarocks.org, so this command will make no difference

I would also like to see a new version deployed to lua rocks.

In the meantime, I think a good first step would be to create a dev rockspec that knows how to build from the repository's source

**Comment by airtonix** [comment]
```
> sudo luarocks install --server=https://luarocks.org/dev lgi
[sudo] password for XYZ:
Installing https://luarocks.org/lgi-0.9.2-1.src.rock

lgi 0.9.2-1 depends on lua >= 5.1 (5.4-1 provided by VM)
make -C lgi
make[1]: Entering directory '/tmp/luarocks_lgi-0.9.2-1-AFaZD8/lgi/lgi'
pkg-config --exists 'gobject-introspection-1.0 >= 0.10.8' --print-errors
touch .depcheck
gcc -fPIC  -O2 -fPIC -I/usr/include -I/usr/include/gobject-introspection-1.0 -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -I/usr/include/sysprof-4 -pthread  -Wall -Wextra -O2 -g -c -o buffer.o buffer.c
gcc -fPIC  -O2 -fPIC -I/usr/include -I/usr/include/gobject-introspection-1.0 -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -I/usr/include/sysprof-4 -pthread  -Wall -Wextra -O2 -g -c -o callable.o callable.c
callable.c: In function ‘closure_callback’:
callable.c:1237:13: error: too few arguments to function ‘lua_resume’
 1237 |       res = lua_resume (L, NULL, npos);
      |             ^~~~~~~~~~
In file included from lgi.h:11,
                 from callable.c:12:
/usr/include/lua.h:300:15: note: declared here
  300 | LUA_API int  (lua_resume)     (lua_State *L, lua_State *from, int narg,
      |               ^~~~~~~~~~
make[1]: *** [Makefile:63: callable.o] Error 1
make[1]: Leaving directory '/tmp/luarocks_lgi-0.9.2-1-AFaZD8/lgi/lgi'
make: *** [Makefile:16: all] Error 2

Error: Build error: Failed building.

```

**Comment by Aire-One** [comment]
@airtonix, we are still in the same boat as last month:

> > just use
> > `sudo luarocks install --server=https://luarocks.org/dev lgi`
> 
> There is no development/scm rockspec uploaded to the lgi module on luarocks.org, so this command will make no difference

Basically, you are installing the release version of LGI, but using Lua 5.4 (your log says `lgi 0.9.2-1 depends on lua >= 5.1 (5.4-1 provided by VM)`) which is only supported by git-master.

> Deploying a new versioned release aside, I think a good first step would be to create a dev rockspec that knows how to build from the repository's source.

This is possible thanks to @leafo work with https://github.com/lgi-devs/lgi/pull/305. You can currently install the LGi git version with the command:

```
luarocks install https://raw.githubusercontent.com/lgi-devs/lgi/master/lgi-scm-1.rockspec
```

**Comment by atticus-sullivan** [comment]
This is fine with me :+1:.

(But a new release including the commits which make `lgi` fit for `lua5.4`, which in fact are already present would still be a good idea, I think. But that's not in my hands)

**Comment by airtonix** [comment]
yep having this on luarocks would be great. imagine all the "OMG BBQ IT DOESNT INSTALL!!!1111pony" tickets you're about to get that you won't have to get. :>

**Comment by Nuc1eoN** [comment]
@psychon Would you mind pushing a new release to LuaRocks with Lua 5.4 support?

**Comment by psychon** [comment]
Why me? @Nuc1eoN Why don't you do it? If I were go guess, I would claim that you are still writing Lua code and use LGI. I do not.

**Comment by Nuc1eoN** [comment]
> Why me? @Nuc1eoN Why don't you do it? If I were go guess, I would claim that you are still writing Lua code and use LGI. I do not.

Excuse me, I was under the impression that you are maintaining the LGI repository.

> Note: I am admin on luarocks.org so I can help get this release sorted out if the original owner is no longer responsive, we can move the lgi name on the root manifest to another account

@leafo Are you still an admin of luarocks.org? Could you still reassign the ownership of that package, so that we can release an official 'dev' version (as you've suggested in #305)? Thanks you.

**Comment by psychon** [comment]
TL;DR: I can try to help with LGI, but don't expect "anything big" from me.

> Excuse me, I was under the impression that you are maintaining the LGI repository.

Heh, in a way I am. But I am not really doing anything. LGI was originally created by @pavouk . At one point in time, I created my own cairo and pango bindings for AwesomeWM and (IIRC) the debian maintainer of AwesomeWM asked me why I am reinventing the wheel instead of just using LGI. That's how I ended up with LGI. In the past, I did some work on this project and I think I am known to (try to) react to bug reports here. But my only "special power" is that I can merge PRs here. I never ever uploaded anything to LuaRocks. And I haven't actually done any "real work" in Lua or for LGI in ages.

Also: Sorry if my answer above was a bit harsh.

**Comment by Frityet** [comment]
> > Why me? @Nuc1eoN Why don't you do it? If I were go guess, I would claim that you are still writing Lua code and use LGI. I do not.
> 
> Excuse me, I was under the impression that you are maintaining the LGI repository.
> 
> > Note: I am admin on luarocks.org so I can help get this release sorted out if the original owner is no longer responsive, we can move the lgi name on the root manifest to another account
> 
> @leafo Are you still an admin of luarocks.org? Could you still reassign the ownership of that package (https://luarocks.org/modules/pavouk/lgi), so that we can release an official 'dev' version (as you've suggested in #305)? Thank you.

@hishamhm

**Comment by Nuc1eoN** [comment]
@psychon

> Heh, in a way I am. But I am not really doing anything. LGI was originally created by @pavouk . At one point in time, I created my own cairo and pango bindings for AwesomeWM and (IIRC) the debian maintainer of AwesomeWM asked me why I am reinventing the wheel instead of just using LGI. That's how I ended up with LGI. In the past, I did some work on this project and I think I am known to (try to) react to bug reports here. But my only "special power" is that I can merge PRs here. I never ever uploaded anything to LuaRocks. And I haven't actually done any "real work" in Lua or for LGI in ages.

I see thank you for the clarificatio! =) Actually well done on keeping this project maintained under those circumstances!!

>Also: Sorry if my answer above was a bit harsh.

No, that's fine all good!

As to the issue, I have looked into it again and it appears that the correct procedure to transfer a project to a new maintainer on LuaRocks is via an issue report on https://github.com/luarocks/luarocks-site/issues.

See https://github.com/luarocks/luarocks-site/issues/205 for instance.

It does not look very complicated and would (at least) allow adding a `git` version to LuaRocks:LGI. That alone would make it worthwhile and easier for lots of people I guess.

Therefore I've considered posting there and to take over the package on LuaRocks, but frankly it does not feel right with me. Actually I am not in a any way associated with LGI. I am not even a lua programmer. I am simply maintaining a package repository on GoboLinux and updating our AwesomeWM setup demands the LGI `git` version (due to Lua 5.4 incompatibilities). To date, we have been using the LuaRocks to install LGI. I lowers our packaging burden. But if that is not possible anymore I will package LGI separately and that should work as well.

To sum it up although I would technically be able to update the LuaRocks package (and even motivated to do that!) for LGI, I just think I am not the right person to take the ownership. And I think you @psychon are the only sensible person to take over the LuaRocks person in all honesty :)

But if you wanna keep it as it is that's fine^^ 

I am just leaving this information here in case you would find it useful :) All the best and thank you so much for your efforts so far!!

**Comment by psychon** [comment]
Thanks for the message. I would then go with not doing anything here. I am not even doing any Lua programming any more.

Let's see if "LGI is unmaintained" makes people appear who want to be involved a bit. If someone appears: I'll try to help where I can, but I cannot even give anyone any permissions on this repo.
